Adyen.Test/HeaderRequestTest.cs

```diff
@@ -30,90 +33,81 @@ namespace Adyen.Test
     [TestClass]
     public class HeaderRequestTest
     {
-
-        private readonly HttpUrlConnectionClient _httpUrlConnectionClient;
         private readonly string _endpoint = "https://endpoint:8080/";
-
-        public HeaderRequestTest()
-        {
-            _httpUrlConnectionClient = new HttpUrlConnectionClient();
-        }
+        
         [TestMethod]
+        [ExpectedException(typeof(InvalidOperationException), "The given header was not found.")]
         public void BasicAuthenticationHeadersTest()
```

Comment:
This is a **very** nice to have, you can open a different ticket for this, if you'd like:

[Enhancement] Rename tests to consistently identify what it-is that you're testing following the "Given _ [MethodName] _ When_ [Condition]_ ResultShouldBe _ [ExpectedResult]" convention.
See: https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#naming-your-tests

This way, it will be clear what is being tested (and you can get rid of unnecessary checks / refactor the unittests into smaller bits, e.g. > do we really need to test the utf-8?)


Example from this PR: `BasicAuthenticationHeadersTest` is testing whether the AuthenticationHeaders are included or not, the same applies to `ApiKeyBasedHeadersTest` where we include "x-api-key" in our headers or not.

Looking at the business logic for this, I expect four tests:
```csharp
"AuthenticationHeaders":     provided | "x-api-key": provided     // Resulting headers["x-api-key"] should be not null
"AuthenticationHeaders":     provided | "x-api-key": not-provided // Result Authentication headers should be not null
"AuthenticationHeaders": not-provided | "x-api-key": provided     // Resulting headers["x-api-key"] should be not null
"AuthenticationHeaders": not-provided | "x-api-key": not-provided // Resulting headers["x-api-key"] should be null, result authentication headers should be null or  even throw an error here as no authentication is provided.
```

Example: `Given_GetHttpWebRequest_When_AuthenticationHeaders_Is_Provided_ResultShouldBe_XXXX`

This covers all cases for the business logic in these lines: https://github.com/Adyen/adyen-dotnet-api-library/pull/690/files#diff-8b602ca15935ec373df692637c85201512137f432ab908df44fa25ad74e8467bR107-R117
```csharp
  if (isApiKeyRequired || _config.HasApiKey)
  {
      httpWebRequest.Headers.Add("x-api-key", _config.XApiKey);
  }
  else if (_config.HasPassword)
  {
      // ...
  }
```